File: packages/ogdf.js/src/layouts/planarity/PlanarizationLayout.js
import { PARAMETER_TYPE } from '../../utils/parameters'
import CrossingMinimizationModule from '../../module/config/CrossingMinimization'
import CCLayoutPackModule from '../../module/config/CCLayoutPack'
import LayoutPlanRepModule from '../../module/config/LayoutPlanRep'
import EmbedderModule from '../../module/config/Embedder'

export default {
    minCliqueSize: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 10
    },
    pageRatio: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 1.0
    },
    crossMin: {
        type: PARAMETER_TYPE.MODULE,
        module: CrossingMinimizationModule,
        default: CrossingMinimizationModule.SubgraphPlanarizer
    },
    embedder: {
        type: PARAMETER_TYPE.MODULE,
        module: EmbedderModule,
        default: EmbedderModule.SimpleEmbedder
    },
    packer: {
        type: PARAMETER_TYPE.MODULE,
        module: CCLayoutPackModule,
        default: CCLayoutPackModule.TileToRowsCCPacker
    },
    planarLayouter: {
        type: PARAMETER_TYPE.MODULE,
        module: LayoutPlanRepModule,
        default: LayoutPlanRepModule.OrthoLayout
    }
}

File: packages/ogdf.js/src/entry/planarity/PlanarizationLayout.cpp
#include <ogdf/planarity/PlanarizationLayout.h>
#include "../main.h"

EM_PORT_API(PlanarizationLayout *)
LayoutModule_PlanarizationLayout(int minCliqueSize, double pageRatio, CrossingMinimizationModule *crossMin, CCLayoutPackModule *packer, EmbedderModule *embedder, LayoutPlanRepModule *planarLayouter)
{
	//LayoutModule
	PlanarizationLayout *model = new PlanarizationLayout();

	model->minCliqueSize(minCliqueSize);
	model->pageRatio(pageRatio);

	model->setCrossMin(crossMin);
	model->setPacker(packer);
	model->setEmbedder(embedder);
	model->setPlanarLayouter(planarLayouter);

	return model;
}
